### PR TITLE
fix ycmd and company-c-header conflict

### PR DESCRIPTION
**Pull request by sukeyisme**
Thank you for contributing to Spacemacs!

Before you submit this pull request, please ensure it is against the develop branch and not master.

This message should be replaced with a description of your change.

Thank you <3

### CODE REVIEW

**Comment by bmag** [comment]
There are two unrelated changes here, please split to two PRs. Also, give some details about the changes - if they are bug fixes, what bugs do they fix and please link to related issues.

**Comment by sukeyisme** [comment]
#8423 8f8a9f2 2a4d4d5
Ingore f2e236  i will split.

File: layers/+lang/c-c++/packages.el [hunk 1]
```diff
@@ -17,7 +17,7 @@
     cmake-mode
     company
     (company-c-headers :toggle (configuration-layer/package-usedp 'company))
-    company-ycmd
+    ;; company-ycmd
```
Comment (deb0ch) [inline layers/+lang/c-c++/packages.el]: Do we want to keep commented code in the code base ?

File: layers/+lang/c-c++/packages.el [hunk 2]
```diff
@@ -128,8 +129,8 @@
     (spacemacs/set-leader-keys-for-major-mode mode
       "gG" 'ycmd-goto-imprecise)))
 
-(defun c-c++/post-init-company-ycmd ()
-  (spacemacs|add-company-backends :backends company-ycmd :modes c-mode-common))
+;; (defun c-c++/post-init-company-ycmd ()
+;;   (spacemacs|add-company-backends :backends company-ycmd :modes c-mode-common))
```
Comment (deb0ch) [inline layers/+lang/c-c++/packages.el]: Idem.

**Comment by sukeyisme** [comment]
fix #1588

**Comment by syl20bnr** [comment]
Thank you for the PR!
I'm sorry but I don't understand the goal of this PR, you removed `company-ycmd` from the package list but still used `(configuration-layer/package-usedp 'company-ycmd)` in the configuration of `post-init-company` which is contradictory.

Did you mean to remove `company-ymcd` completely ? If yes, why ?

**Comment by sukeyisme** [comment]
Not to remove, please see the problem #1588.

**Pull request closed without merging**